tests/transforms/test_split_non_commuting.py

```diff
@@ -122,765 +119,580 @@ def complex_wires_processing_fn(results):
-            for i in range(3):
-                assert qml.math.allclose(res[i], expected_results, atol=0.05)
-        else:
-            assert qml.math.allclose(res, expected_results, atol=0.05)
+        shot_dimension = (3,) if isinstance(shots, list) else ()
+        parameter_dimension = (2,) if len(qml.math.shape(params)) > 1 else ()
+        expected_dimension = shot_dimension + parameter_dimension
+        assert qml.math.shape(res) == expected_dimension
 
-    @pytest.mark.parametrize("grouping_strategy", [None, "default", "qwc", "wires"])
-    @pytest.mark.parametrize("shots", [20000, [20000, 30000, 40000]])
+    @pytest.mark.parametrize("grouping_strategy", [None, "qwc", "wires"])
+    @pytest.mark.parametrize("shots", [10, [10, 20, 30]])
     @pytest.mark.parametrize(
-        "params, expected_results",
+        "params",
         [
-            (
-                [np.pi / 4, 3 * np.pi / 4],
-                [
-                    0.5,
-                    -0.5 * np.cos(np.pi / 4),
-                    0.5 + np.cos(np.pi / 4) * 0.5 + 2.0 * 0.5 + 1,
-                    np.dot(
-                        [0.1, 0.2, 0.3, 0.4],
-                        [-0.5, -0.5 * np.cos(np.pi / 4), 0.5 * np.cos(np.pi / 4), 1],
-                    ),
-                    1.5,
-                ],
-            ),
-            (
-                [[np.pi / 4, 3 * np.pi / 4], [3 * np.pi / 4, 3 * np.pi / 4]],
-                [
-                    [0.5, -0.5],
-                    [-0.5 * np.cos(np.pi / 4), -0.5 * np.cos(np.pi / 4)],
-                    [
-                        0.5 + np.cos(np.pi / 4) * 0.5 + 2.0 * 0.5 + 1,
-                        -0.5 - np.cos(np.pi / 4) * 0.5 - 2.0 * 0.5 + 1,
-                    ],
-                    [
-                        np.dot(
-                            [0.1, 0.2, 0.3, 0.4],
-                            [-0.5, -0.5 * np.cos(np.pi / 4), 0.5 * np.cos(np.pi / 4), 1],
-                        ),
-                        np.dot(
-                            [0.1, 0.2, 0.3, 0.4],
-                            [0.5, 0.5 * np.cos(np.pi / 4), -0.5 * np.cos(np.pi / 4), 1],
-                        ),
-                    ],
-                    [1.5, 1.5],
-                ],
-            ),
+            [0.1, 0.2, 0.3],
+            [[0.1, 0.1], [0.2, 0.2], [0.3, 0.3]],
         ],
     )
-    def test_mixed_measurement_types(
-        self, grouping_strategy, shots, params, expected_results, seed
-    ):
-        """Tests that a QNode with mixed measurement types is executed correctly"""
+    def test_general_circuits(self, grouping_strategy, shots, params):
+        """Tests executing a QNode with different grouping strategies on a typical circuit."""
 
-        dev = qml.device("default.qubit", wires=2, shots=shots, seed=seed)
-
-        obs_list = complex_obs_list
+        dev = qml.device("default.qubit", wires=3, shots=shots)
 
+        @partial(split_non_commuting, grouping_strategy=grouping_strategy)
```

Comment:
Same as above --- do we need an execution to know this is correct?

Reply:
I feel that some integration testing with execution is needed, just to make sure that everything works, especially with different return shapes and types associated with parameter broadcasting and shot vectors.

Reply:
Same as above --- as this is a system (end-to-end) rather than integration test, we can come back later and mock out the device execution to see the interplay with the qnode only.